contracts/FeeDisbursal/FeeDisbursal.sol

```diff
@@ -0,0 +1,265 @@
+    /// @param token address of the claimable token
+    /// @param until until what index to claim to
+    function claim(address token, uint256 until)
+    public
+    returns (uint256 claimableAmount, uint256 lastIndex)
+    {
+        // only allow verified and active accounts to claim tokens
+        // @TODO: move access control to Controller
+        IdentityClaims memory claims = deserializeClaims(IDENTITY_REGISTRY.getClaims(msg.sender));
+        require(claims.isVerified && !claims.accountFrozen, "NF_DISB_NOT_VER");
+        return claimPrivate(token, msg.sender, until);
+    }
+
+    /// @notice claim multiple tokens, to be called an investor
+    /// @param tokens addresses of the claimable token
+    function claimMultiple(address[] tokens)
+    public
+    {
+        // only allow verified and active accounts to claim tokens
+        // @TODO: move access control to Controller
+        IdentityClaims memory claims = deserializeClaims(IDENTITY_REGISTRY.getClaims(msg.sender));
+        require(claims.isVerified && !claims.accountFrozen, "NF_DISB_NOT_VER");
+        for (uint256 i = 0; i < tokens.length; i += 1) {
+            claimPrivate(tokens[i], msg.sender, UINT256_MAX);
+        }
+    }
+
+    /// @notice check how many tokens of a certain kind can be claimed by an account
+    /// @param token address of the claimable token
+    /// @param spender address of the spender that would receive the funds
+    /// @param until until what index to claim to
+    function claimable(address token, address spender, uint256 until)
+    public
+    view
+    returns (uint256 claimableAmount, uint256 lastIndex)
+    {   
+        // we don't do to a verified check here, this serves purely to check how much is claimable for an address
+        return claimablePrivate(token, spender, until);
+    }
+
+    /// @notice claim a token, to be called an investor
+    /// @param tokens addresses of the claimable token
+    /// @param spender address of the spender that would receive the funds
+    function claimableMutiple(address[] tokens, address spender)
+    public
+    view
+    returns (uint256[])
+    {   
+        // we don't to do a verified check here, this serves purely to check how much is claimable for an address
+        uint256[] memory result = new uint256[](tokens.length);
+        for (uint256 i = 0; i < tokens.length; i += 1) {
+            (uint256 claimableAmount, uint256 lastIndex) = claimablePrivate(tokens[i], spender, UINT256_MAX);
+            result[i] = claimableAmount;
+        }
+        return result;
+    }
+
+    /// @notice claim a token, to be called an investor
+    /// @param token address of the recyclable token
+    /// @param investors list of investors we want to recycle tokens for
+    /// @param until until what index to claim to
+    function recycle(address token, address[] investors, uint256 until)
+    public
+    {        
+        // @TODO: Recycle funds
```

Comment:
My main questions for this are:
- Do we create a new disbursal entry for every call of this function? Or do we accumlate the tokens and then disburse if there is a certain amount.
- Does the pro-rata token entry need to be retained, or do we default to NEU? If we have to retain the pro rata token, we can only do a redisbursal per original disbursal. 
- In any case, some tokens will be locked in limbo forever. Without saving which tokens have been claimed so far, we will never be able to get them back.

Reply:
ad 1. if you implement merging latest disbursals with current snapshot id then this problem is gone
ad 2. hmmm I do not get it. you do disbursal per token originally made.
ad 3. why? you can recover past balances of the shareToken to get who didn't claim yet... not easy but can be done.